var/spack/repos/builtin/packages/py-gevent/package.py

```diff
@@ -39,6 +39,15 @@ class PyGevent(PythonPackage):
     # Deprecated compiler options. upstream PR: https://github.com/gevent/gevent/pull/1896
     patch("icc.patch", when="%intel")
 
+    @run_before("install")
+    def recythonize(self):
+        # Clean pre-generated cython files -- we've seen issues with Python 3.8 due to
+        # an old cython that was used to generate the C sources.
+        # On top of that, they specify a prerequisite on a file in cython's prefix,
+        # meaning that cython runs again depending on whether it was installed before e.g.
+        # 2020... So, just clean and re-run from scratch instead.
+        python("setup.py", "clean")
```

Comment:
Direct invocation of setup.py is deprecated: https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html

I would be in favor of simply adding upper bounds on Python versions to the package. I think there's also a way to manually remove cython files that other packages use.